### PR TITLE
⭐️New: Add `vue/no-use-v-if-with-v-for` rule

### PR DESCRIPTION
**Pull request by ota-meshi**
This PR adds `vue/no-use-v-if-with-v-for` rule.
This implements rule proposed in #403

### CODE REVIEW

File: docs/rules/no-use-v-if-with-v-for.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+`allowUsingIterationVar` - Enables The `v-if` directive use the reference which is to the variables which are defined by the `v-for` directives.
+
+```js
+'vue/no-use-v-if-with-v-for': ['error', {
+  allowUsingIterationVar: true // default: false
+}]
+```
+
+:+1: Examples of additional **correct** code for this rule with sample `"allowUsingIterationVar": true` options:
```
Comment (michalsnik) [inline docs/rules/no-use-v-if-with-v-for.md]: I guess this is supposed to be `incorrect` right?

File: docs/rules/no-use-v-if-with-v-for.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+```html
+<TodoItem
+  v-if="complete"
+  v-for="todo in todos"
+  :todo="todo"
+/>
+```
+
+:-1: Examples of additional **incorrect** code for this rule with sample `"allowUsingIterationVar": true` options:
```
Comment (michalsnik) [inline docs/rules/no-use-v-if-with-v-for.md]: And this correct

File: lib/rules/no-use-v-if-with-v-for.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      "VAttribute[directive=true][key.name='if']" (node) {
+        const element = node.parent.parent
+
+        if (utils.hasDirective(element, 'for')) {
+          if (isUsingIterationVar(node)) {
+            if (!allowUsingIterationVar) {
+              context.report({
+                node,
+                loc: node.loc,
+                message: "The 'v-for' list variable should be replace with a new computed property that returns your filtered list by this 'v-if' condition."
```
Comment (michalsnik) [inline lib/rules/no-use-v-if-with-v-for.js]: Can we make `list` to be dynamic value in this message? Also what do you think about this kind of message?
```
The 'someArray' variable inside 'v-for' directive should be replaced with a computed property that returns filtered array instead. You should not mix 'v-for' with 'v-if'.
```

**Comment by ota-meshi** [comment]
@michalsnik Thank you for review!
I changed it, so please check it.

**Pull request merged**